apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommandTextBuilder.cs

```diff
@@ -276,5 +279,194 @@ private static string ValidateTableName(string table, string parameterName)
             }
             return table;
         }
+
+        private static readonly char s_singleQuote = '\'';
+        private static readonly char s_doubleQuote = '"';
+        private static readonly char s_backtickQuote = '`';
+        private static readonly char s_hyphen = '-';
+        private static readonly char s_hash = '#';
+        private static readonly char s_slash = '/';
+        private static readonly char s_asterisk = '*';
+
+        /// <summary>
+        /// Removes any comments from the given sql string.
+        /// </summary>
+        /// <param name="sql">The sql string to strip for comments</param>
+        /// <returns>The sql string without any comments</returns>
+        /// <exception cref="ArgumentException">
+        /// Throws ArgumentException if the sql string contains an unclosed
+        /// literal or unclosed quoted identifier
+        /// </exception>
+        internal static string RemoveComments(string sql)
```

Comment:
OK, I think we can make this simpler as well:
- We have 2 kinds of comments, line comments, starting with `--` for instance, and block comments `/* */`. And we only care about those at the beginning of the command.

In a loop:
- If the command string starts with any of the line comment starters `--`, `#`, or others, remove the entire line and loop back.
- If the commnd string starts with any of the block comment starterts, `/*` (others?) just find the first comment block closer that's not escaped `*/`.
  -  If found, remove everything up to and including the closer `*/` and loop back again.
  - Else, return the original string and let the caller or the backend fail.
- Return what you have when it doesn't start with a line or block comment.

WDTY? Or am I missing something?

Reply:
I think this will break because line comments can also start in the middle of a line, which would not be stripped by such a simplified approach. And I worry that those 'inline line comments' that would be left in the command string could break the rest of the parsing. So the following kind of statements would probably not be parsed correctly:

```sql
@{
  OPTIMIZER_VERSION=1 -- Use the first optimizer for now. Change this to @{OPTIMIZER_VERSION=2} once XYZ is done.
}
SELECT *
FROM Singers

@{OPTIMIZER_VERSION=1} -- This is a comment /* this is not a block comment
SELECT *
FROM Singers
/* This is a block comment */
```

The current statement hint parsing is simplified to just look for the first and next curly brace, but that can only be safely done because we at that moment know that:
1. String literals cannot occur in that part of a valid SQL string.
2. All comments in that part of the SQL string have been removed.

> `/*` (others?)

No, `/*` is the only valid start of a block comment.